src/pygambit/behav.pxi

```diff
@@ -284,8 +284,13 @@ class MixedBehaviorProfile:
             If `player` is a `Player` from a different game.
         KeyError
             If `player` is a string and no player in the game has that label.
+        ValueError
+            If `player` resolves to the chance player
         """
-        return self._payoff(self.game._resolve_player(player, 'payoff'))
+        resolved_player = self.game._resolve_player(player, 'payoff')
+        if resolved_player.is_chance:
+            raise ValueError("It is not possible to call payoff for the chance player")
```

Comment:
I would suggest saying "payoff() is not defined for the chance player".  After all it's possible to call the function (the calling code just did so!).  It's just that it's not a well-defined concept.  Similar comment on the other functions.